### PR TITLE
Add additional validations for call against system account

### LINKED ISSUE #7621: Executing eth_call with value against a system account returns success
### Description

Executing an eth_call against a system account(0.0.444) with amount results in status 200 and 0x as response.

### Steps to reproduce

1. Execute the following request:
{
    "from": "0x0000000000000000000000000000000000000002",
    "estimate": true,
    "to": "0x00000000000000000000000000000000000001bc",
    "value": 933333
}

**Expected result**: The API should return status 400 with invalid fee submited
**Actual result**: The API returns 200 with 0x as response.

### Additional context
This is also reproducible when we do it via internal call using a contract as middleman.

 function makeCallWithAmount(address _to, bytes memory _data) external payable returns (bool success, bytes memory returnData) {
        (success, returnData) = _to.call{value: msg.value}(_data);
    }
 
_No response_

### Hedera network

other

### Version

0.96.0

### Operating system

None

### PR DESCRIPTION
**Pull request by zhpetkov**
**Description**:
Add validations in `AbstractEvmMessageCallProcessor` which checks the type of the account and value for the transaction - cannot send value to native precompile calls, since there are collisions with the system account and value will be transferred to the system account, which is undesired. The biggest part of the implementation is from Hedera Services. 

**Related issue(s)**: 

Fixes #7621

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/AbstractEvmMessageCallProcessor.java [hunk 1]
```diff
@@ -16,40 +16,50 @@
...
+import static com.hedera.services.stream.proto.ContractActionType.SYSTEM;
+import static org.apache.tuweni.bytes.Bytes.EMPTY;
+import static org.hyperledger.besu.evm.frame.ExceptionalHaltReason.DefaultExceptionalHaltReason.INSUFFICIENT_GAS;
+import static org.hyperledger.besu.evm.frame.MessageFrame.State.COMPLETED_SUCCESS;
+import static org.hyperledger.besu.evm.frame.MessageFrame.State.EXCEPTIONAL_HALT;
+import static org.hyperledger.besu.evm.frame.MessageFrame.State.REVERT;
+
 public abstract class AbstractEvmMessageCallProcessor extends HederaEvmMessageCallProcessor {
 
+    private final int RESERVED_HTS_SYSTEM_ACCOUNT = 359;
```
Comment (bilyana-gospodinova) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/AbstractEvmMessageCallProcessor.java:49]: There is no need to define the constant again. It is available in `EvmHTSPrecompiledContract#EVM_HTS_PRECOMPILED_CONTRACT_ADDRESS`.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/AbstractEvmMessageCallProcessor.java [hunk 2]
```diff
@@ -64,6 +74,22 @@ public void start(final MessageFrame frame, final OperationTracer operationTrace
                 && operationTracer instanceof HederaOperationTracer hederaOperationTracer) {
             hederaOperationTracer.tracePrecompileResult(frame, isHederaPrecompile ? SYSTEM : PRECOMPILE);
         }
+
+        final boolean nonHTS = Integer.compareUnsigned(contractAddress.getInt(16), RESERVED_HTS_SYSTEM_ACCOUNT) != 0;
+
+        if (!isNativePrecompile &&
```
Comment (bilyana-gospodinova) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/AbstractEvmMessageCallProcessor.java:80]: Please leave a comment why we need this distinction between an HTS and a non-HTS address. I suppose the reason is that an HTS token create call can receive value.

**Review by bilyana-gospodinova — CHANGES_REQUESTED** [review]
Please add new tests that cover the changes added in `AbstractEvmMessageCallProcessor`.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorTest.java [hunk 3]
```diff
@@ -135,4 +143,90 @@ void startWithNonPrecompileAddressDoesNotTracePrecompileResult() {
...
+    @Test
+    void startWithNonHTSAndWithoutValuePrecompiledContractAddress() {
+        subject = new MirrorEvmMessageCallProcessor(
+                autoCreationLogic, entityAddressSequencer, evm, precompiles, precompilesHolder, gasCalculatorHederaV22, address -> true);
+        var contractAddress = Address.fromHexString(EVM_HTS_PRECOMPILED_CONTRACT_ADDRESS);
+        var recAddress = Address.fromHexString(EVM_HTS_PRECOMPILED_CONTRACT_ADDRESS);
+        when(messageFrame.getContractAddress()).thenReturn(contractAddress);
+        when(messageFrame.getRecipientAddress()).thenReturn(recAddress);
+        when(messageFrame.getValue()).thenReturn(Wei.of(0L));
+        when(messageFrame.getCurrentOperation()).thenReturn(createOperation());
```
Comment (edwin-greene) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorTest.java:195]: Test seems to pass with line 195 commented out. Can it be removed or is there something more that should be verified?

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorTest.java [hunk 4]
```diff
@@ -135,4 +143,90 @@ void startWithNonPrecompileAddressDoesNotTracePrecompileResult() {
...
+    @Test
+    void startWithHTSPrecompiledContractAddress() {
+        subject = new MirrorEvmMessageCallProcessor(
+                autoCreationLogic, entityAddressSequencer, evm, precompiles, precompilesHolder, gasCalculatorHederaV22, address -> true);
+        var contractAddress = Address.fromHexString(EVM_HTS_PRECOMPILED_CONTRACT_ADDRESS);
+        var recAddress = Address.fromHexString(EVM_HTS_PRECOMPILED_CONTRACT_ADDRESS);
+        when(messageFrame.getContractAddress()).thenReturn(contractAddress);
+        when(messageFrame.getRecipientAddress()).thenReturn(recAddress);
+        when(messageFrame.getValue()).thenReturn(Wei.of(1000L));
+        when(messageFrame.getCurrentOperation()).thenReturn(createOperation());
```
Comment (edwin-greene) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorTest.java:178]: Test seems to pass with line 178 commented out. Can it be removed or is there something more that should be verified?

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorTest.java [hunk 5]
```diff
@@ -135,4 +143,90 @@ void startWithNonPrecompileAddressDoesNotTracePrecompileResult() {
...
+        var recAddress = Address.fromHexString(EVM_HTS_PRECOMPILED_CONTRACT_ADDRESS);
+        when(messageFrame.getContractAddress()).thenReturn(contractAddress);
+        when(messageFrame.getRecipientAddress()).thenReturn(recAddress);
+        when(messageFrame.getSenderAddress()).thenReturn(recAddress);
+        when(autoCreationLogic.create(any(), any(), any(), any(), any())).thenReturn(Pair.of(OK, 1000L));
+        when(messageFrame.getValue()).thenReturn(Wei.of(5000L));
+        when(messageFrame.getBlockValues()).thenReturn(new HederaBlockValues(0L, 0L, Instant.EPOCH));
+        when(messageFrame.getGasPrice()).thenReturn(Wei.ONE);
+        when(opcodeTracer.getContext().getOpcodeTracerOptions()).thenReturn(new OpcodeTracerOptions(true, true, true));
+        when(messageFrame.getCurrentOperation()).thenReturn(createOperation());
```
Comment (edwin-greene) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorTest.java:161]: This seems to be the only line that uses `createOperation`. To reduce the lines of code, could `createOperation` be removed and replaced with a mock?

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by bilyana-gospodinova — APPROVED** [review]
LGTM

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorTest.java [hunk 6]
```diff
@@ -135,4 +144,88 @@ void startWithNonPrecompileAddressDoesNotTracePrecompileResult() {
...
+        when(messageFrame.getValue()).thenReturn(Wei.of(0L));
+
+        subject.start(messageFrame, opcodeTracer);
+
+        verify(opcodeTracer).tracePrecompileResult(messageFrame, ContractActionType.SYSTEM);
+        assertNull(messageFrame.getState());
+    }
+
+    @NotNull
+    private static Operation createOperation() {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmMessageCallProcessorTest.java:203]: Probably obsolete method

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**